test/ruby/test_ast.rb

```diff
@@ -154,19 +155,10 @@ def test_parse_raises_syntax_error
   end
 
   def test_parse_file_raises_syntax_error
-    Tempfile.create(%w"test_ast .rb") do |f|
+    Tempfile.create("test_ast.rb") do |f|
```

Comment:
What is the intention of this change?

Reply:
If the question is regarding that specific line, I assumed the space in the filename was a typo, and I couldn't identify any purpose to the `%w`, but it made the test harder to understand.

If the question is about the commit altogether, I was looking at the code because I had hit the bug it fixes, but couldn't reproduce it on the trunk branch, so I wanted to see how it was fixed. I found the commit, but got really confused by this test for about 30 minutes (eg "why the space in the filename, why the `%w`, why use quotes for `%w`'s delimiter but brackets for `%W`'s delimiter, is the dash on the subprocess tellilng it to fork Ruby (like `popen`) or to read from stdin (like Ruby's command-line args) or is it delimiting interpreter args from program args, why pass it as an arg instead of interpolating it, why 2 heredocs, etc). After thinking about it for that long, and going through the refactorings in my mind, I concluded that it could achieve its goal in a much simpler manner, which is what I've submitted.

I think the reason it went through a subprocess is because it was previously setting a global error instead of raising it (the err is not printed by the rescue block, but rather by Ruby, which sets the exit status to 1 as a consequence)

![image](https://user-images.githubusercontent.com/77495/45038640-41442c00-b027-11e8-9380-54120d9e4b32.png)

But I don't think it needs to worry about failing gracefully with the old buggy behaviour it only needs to worry about failing if the new correct behaviour is not expressed. This version of the test will do that since the old behaviour did not raise the error.

Reply:
The old line didn't make a file with a space in the filename:

```ruby
>> Tempfile.create(%w"test_ast .rb") { |f| f.path }
"/var/folders/dx/6tq721rs17bcplgwmtltyn5nqmv_ww/T/test_ast20180905-7282-1ik1z18.rb"
```
```ruby
>> Tempfile.create("test_ast.rb") { |f| f.path }
"/var/folders/dx/6tq721rs17bcplgwmtltyn5nqmv_ww/T/test_ast.rb20180905-7282-11dgzd"
```
Rather, the array is used as the filename's prefix and extension, respectively.

Double quotes were probably a poor choice of delimiter there. I'll grant you that.

Reply:
Ahh, I mixed it up with `%q`, I'll change it to `["test", ".rb"]`